site/src/hooks/useClipboard.test.tsx

```diff
@@ -133,88 +135,120 @@ function renderUseClipboard(inputs: UseClipboardInput) {
   );
 }
 
-type ScheduleConfig = Readonly<{ isHttps: boolean }>;
+const secureContextValues: readonly boolean[] = [true, false];
 
-export function scheduleClipboardTests({ isHttps }: ScheduleConfig) {
-  const mockClipboardInstance = makeMockClipboard(isHttps);
-  const originalNavigator = window.navigator;
+// Not a big fan of describe.each most of the time, but since we need to test
+// the exact same test cases against different inputs, and we want them to run
+// as sequentially as possible to minimize flakes, they make sense here
+describe.each(secureContextValues)("useClipboard - secure: %j", (isSecure) => {
+  const {
+    mockClipboard,
+    mockExecCommand,
+    getClipboardText,
+    setSimulateFailure,
+    resetClipboardMocks,
+  } = setupMockClipboard(isSecure);
 
   beforeEach(() => {
     jest.useFakeTimers();
+    global.document.execCommand = mockExecCommand;
     jest.spyOn(window, "navigator", "get").mockImplementation(() => ({
       ...originalNavigator,
-      clipboard: mockClipboardInstance,
+      clipboard: mockClipboard,
     }));
 
-    if (!isHttps) {
-      // Not the biggest fan of exposing implementation details like this, but
-      // making any kind of mock for execCommand is really gnarly in general
-      global.document.execCommand = jest.fn(() => {
-        if (mockClipboardInstance.simulateFailure) {
-          return false;
-        }
-
-        const dummyInput = document.querySelector("input[data-testid=dummy]");
-        const inputIsFocused =
-          dummyInput instanceof HTMLInputElement &&
-          document.activeElement === dummyInput;
-
-        let copySuccessful = false;
-        if (inputIsFocused) {
-          mockClipboardInstance.setMockText(dummyInput.value);
-          copySuccessful = true;
-        }
-
-        return copySuccessful;
-      });
-    }
+    console.error = (errorValue, ...rest) => {
+      const canIgnore =
+        errorValue instanceof Error &&
+        errorValue.message === COPY_FAILED_MESSAGE;
+
+      if (!canIgnore) {
+        originalConsoleError(errorValue, ...rest);
+      }
+    };
```

Comment:
is there a reason we can't use `spyOn` or something? feels like jest could be helping with this

Reply:
Oh, yeah, good call – just wasn't thinking straight when I wrote that. Just fixed thins